src/interpreter/mod.rs

```diff
@@ -979,10 +994,23 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
         match value {
             Value::ByRef(ptr) => self.move_(ptr, dest, dest_ty),
             Value::ByVal(primval) => self.memory.write_primval(dest, primval),
+            Value::ByValPair(a, b) => {
+                self.memory.write_primval(dest, a)?;
+                let extra_dest = dest.offset(self.memory.pointer_size() as isize);
```

Comment:
Sounds good for now. We might use `ByValPair` as an optimization to avoid allocations for checked math later (which return 2-tuples), but we can adjust this when we add that.